db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1922 @@
+              """
+          );
+
+          // we expect only 1 result
+          assertTrue(res.next());
+
+          // res.getArray() doesn't give you a String[] array - have to do some manipulation to get its length as a result
+          System.out.println(res.getArray("sources").toString());
+          assertEquals(4, res.getArray("sources").toString().split("\",\"").length);
+          assertEquals(7, res.getInt("derived_total"));
+
+          // we expect only 1 result
+          assertFalse(res.next());
+        });
+      }
+    }
+  }
+  //endregion
+
+
+  //region Derivation Tests
+  /**
+   * The focus of this class is to test specifically the derivation function for events. This includes overlapping of
+   *    source windows, as well as reconciling differences inside the sources themselves (by testing each of the 4
+   *    derivation rules).
+   */
+  @Nested
+  @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+  class DerivedEventsTests {
+
+    /**
+     * This class focuses on testing sources' windows to verify that derivation will work as expected. These are
+     *    separate, extracted tests that don't really evaluate events and instead just that sources play together
+     *    correctly. As such we test "empty" sources to make sure their overlapped windows work correctly.
+     * We add events that span a very short duration simply so that the sources show up in derived_events, but we aren't
+     *    testing any properties of said events.
+     */
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class DerivedSourcesTests {
+
+      // Commonly Repeated:
+      String duration = "00:00:00.000001";
+
+      /**
+       * The first test is a basic, non-overlapping case. We must ensure that gaps are preserved:
+       *
+       *    A:             ++++++++
+       *    B:  +++++++
+       *        BBBBBBB    AAAAAAAA
+       */
+      @Test
+      void testSparseCoverage() throws SQLException {
+        try(final var statement = connection.createStatement()) {
+
+          // create our sources and their per-window events
+          ExternalSource a = new ExternalSource(
+              "A",
+              st,
+              dg,
+              "2024-01-01T00:00:00Z",
+              "2024-01-01T03:00:00Z",
+              "2024-01-01T04:00:00Z",
+              ca,
+              mt
+          );
+          ExternalEvent aE = new ExternalEvent(a.key() + "_event", st, a.key(), dg, a.start_time(), duration, mt);
+          ExternalSource b = new ExternalSource(
+              "B",
+              st,
+              dg,
+              "2024-01-02T00:00:00Z",
+              "2024-01-01T01:00:00Z",
+              "2024-01-01T02:00:00Z",
+              ca,
+              mt
+          );
+          ExternalEvent bE = new ExternalEvent(b.key() + "_event", st, b.key(), dg, b.start_time(), duration, mt);
+
+          // verify the ranges are as expected
+          assertDoesNotThrow(() -> {
```

Comment:
You don't need to `assertDoesNotThrow` on this block.

Reply:
Unresolving because, while it was addressed here, this issue occurs in multiple places throughout the file.

Reply:
Now resolved.